src/models/logits.cpp

```diff
@@ -188,19 +236,41 @@ DeviceSpan<float> Logits::Get() {
 
     auto batched_logits_cpu = cpu_span<float>{cpu_tensor, element_count};
     HandleEOSArray(batched_logits_cpu);
+    if (!logits_masks_.empty()) {
+      AddMask(batched_logits_cpu, logits_masks_);
+    }
 
     logits_ = WrapTensor<float>(*state_.params_->p_device, *value32_cpu_);
     return logits_;
   }
 #endif
 
   HandleEOSArray(logits_.Span());
+  if (!logits_masks_.empty()) {
+    AddMask(logits_.Span(), logits_masks_);
+  }
   return logits_;
 }
 
 #pragma warning(pop)
 
-void Logits::Update() {
+void Logits::Update(DeviceSpan<int32_t> next_tokens_unk) {
+  if (!logits_processors_.empty() && logits_processors_.at(0)) {
+    auto next_tokens = next_tokens_unk.CopyDeviceToCpu();
+    for (int i = 0; i < next_tokens.size(); i++) {
+      logits_processors_[i]->CommitTokens(static_cast<uint32_t>(next_tokens[i]));
+    }
+    mask_future_ = std::async(std::launch::async, [&]() {
```

Comment:
Should previous `future` be released/stopeed before assigning it a new `future`?

Reply:
future.get() will wait future to finish. And it will also throw the exception in aysnc running. So I think it may not need to release here.

Reply:
good. thanks for the explaination.